akka-persistence-typed/src/main/scala/akka/persistence/typed/scaladsl/PersistentBehaviors.scala

```diff
@@ -98,6 +98,11 @@ trait PersistentBehavior[Command, Event, State] extends DeferredBehavior[Command
    */
   def onRecoveryCompleted(callback: (ActorContext[Command], State) ⇒ Unit): PersistentBehavior[Command, Event, State]
 
+  /**
+   * Change the recovery strategy
+   */
+  def withRecovery(recovery: Recovery): PersistentBehavior[Command, Event, State]
```

Comment:
I think we should define a new `Recovery` in `akka.persistence.typed`. We might not want to support all of the recovery strategies. For example I'm skeptical to `toSequenceNr` and `replayMax`.

Remove the `import akka.persistence._` in this file and replace with full class imports so we see the dependencies more clearly.